uvtools/dspec.py

```diff
@@ -294,21 +380,53 @@ def linear_delay_filter(data, wgts, df, filter_centers, filter_widths, filter_fa
 
     output = np.zeros_like(data)
     nchan = data.shape[1]
-    for sample_num, sample, wght in zip(range(d_shape[0]), data, wgts):
-        wght_mat = np.outer(wght.T, wght)
-        filter_mat = sinc_downweight_mat_inv(nchan, df, filter_centers, filter_widths, filter_factors, cache) * wght_mat
-        filter_key = (nchan, df, ) + tuple(filter_centers) + \
-        tuple(filter_widths) + tuple(filter_factors) + tuple(wght.tolist()) + ('inverse',)
-        if not filter_key in cache:
-            cache[filter_key] = np.linalg.pinv(filter_mat)
-        filter_mat = cache[filter_key]
-        #print(np.std(sample.real))
-        #print(np.diag(filter_mat))
-        output[sample_num] = np.dot(filter_mat, sample)
-        #print(np.std(output[sample_num].real))
-    if data_1d:
-        output = output[0]
-    return output
+    ntimes = data.shape[0]
+    info = {'filter_centers':filter_centers, 'filter_widths':filter_widths, 'filter_factors': filter_factors}
+    skipped_0 = []
+    skipped_1 = []
+    if clean_dimensions[1]:
```

Comment:
it would be helpful to add at least a few comments over the next dozen or so lines to describe what you are doing, in case someone else needs to update this code

Reply:
I've added some comments here and cleaned up the code :)